src/corona_hakab_model/medical_state.py

```diff
@@ -9,35 +9,20 @@ class MedicalState(State, ABC):
     test_willingness: float
     detectable: bool
 
-    def val(self):
-        return self.agent_count
-
-
-class SusceptibleState(MedicalState, ABC):
-    def __init__(self, *args, test_willingness: float, **kwargs):
+    def __init__(self, *args, **kwargs):
+        self.susceptible = kwargs.pop('susceptible')
+        self.contagiousness = kwargs.pop('contagiousness')
+        self.test_willingness = kwargs.pop('test_willingness')
+        self.detectable = kwargs.pop('detectable')
         super().__init__(*args, **kwargs)
-        self.test_willingness = test_willingness
-
-    contagiousness = 0
-    susceptible = True
-    detectable = False
 
+    def val(self):
+        return self.agent_count
 
-class ContagiousState(MedicalState, ABC):
-    susceptible = False
-    detectable = True
 
-    def __init__(self, *args, contagiousness: float, test_willingness: float, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.test_willingness = test_willingness
-        self.contagiousness = contagiousness
+class TerminalMedicalState(MedicalState, TerminalState):
+    pass
 
 
-class ImmuneState(MedicalState, ABC):
-    susceptible = False
-    contagiousness = 0
-
-    def __init__(self, *args, detectable: bool, test_willingness: float, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.test_willingness = test_willingness
-        self.detectable = detectable
+class StochasticMedicalState(MedicalState, StochasticState):
```

Comment:
We should maintain the distinction betweel a MedicalState - which holds information about the disease - and a State (can be named better as MechanicalState) that is defined by transfers and durations.
this is why I wanted there to me a field in the JSON about what the "MedicalState" type is and the "MechanicalState" type is

Reply:
I've renamed the JSON field from `medical_state_type` as you suggested to `mechanism` - hopefully this reduce the confusion.